### PR TITLE
[ROCm] Add AMD_SERIALIZE_KERNEL environment for gpu_offloading_test

### PR DESCRIPTION
**Pull request by hsharsha**
This ensures kernel lauch are serialized.

### CODE REVIEW

**Review by golechwierowicz — APPROVED** [review]

File: xla/service/gpu/BUILD [hunk 1]
```diff
@@ -3626,6 +3626,7 @@ xla_test(
     name = "gpu_offloading_test",
     srcs = ["gpu_offloading_test.cc"],
     backends = ["gpu"],
+    env = if_rocm_is_configured({"AMD_SERIALIZE_KERNEL" : "3"}),
```
Comment (golechwierowicz) [inline xla/service/gpu/BUILD:3629]: Actually this change does not work in non-ROCM paths. They will evaluate to [] which is not a valid.

https://github.com/openxla/xla/blob/main/third_party/tsl/third_party/gpus/rocm/build_defs.bzl.tpl#L55-L63

Can you come up with another solution?

**Review by golechwierowicz — CHANGES_REQUESTED** [review]
Requesting changes, breaks non-ROCM path

**Comment by hsharsha** [comment]
> Requesting changes, breaks non-ROCM path

I am trying out a more fine grained wait for copy-start to wait until the computation is available before starting async copy-start

**Comment by hsharsha** [comment]
This is addressed by https://github.com/openxla/xla/pull/13597/files

**Pull request closed without merging**